gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -101,106 +119,84 @@ def get_vds_for_freq(
+    vds = get_gnomad_v4_vds(
+        test=use_test_dataset,
+        release_only=True,
+        filter_partitions=test_partitions,
+        chrom=chrom,
+        annotate_meta=True,
+    )
 
-        elif test_n_partitions:
-            logger.info(
-                "Filtering to %s partitions for testing purposes...", test_n_partitions
+    if test_gene:
+        logger.info("Filtering to DRD2 in VDS for testing purposes...")
+        test_interval = [
+            hl.parse_locus_interval(
+                "chr11:113409605-113475691", reference_genome="GRCh38"
             )
-            test_vmt = vds.variant_data._filter_partitions(range(test_n_partitions))
-            test_rmt = vds.reference_data._filter_partitions(range(test_n_partitions))
-            vds = hl.vds.VariantDataset(test_rmt, test_vmt)
-        else:
-            logger.info("Filtering to chromosome %s...", chrom)
-            vds = hl.vds.filter_chromosomes(vds, keep=chrom)
-
-    (
-        variants,
-        samples,
-    ) = vds.variant_data.count()  # TODO: Remove this logger after tests?
-    logger.info("VDS has %s variants in %s samples...", variants, samples)
+        ]
+        vds = hl.vds.filter_intervals(vds, test_interval, split_reference_blocks=True)
 
     logger.info("Annotating VDS with only necessary sample metadata...")
     vds = hl.vds.VariantDataset(
         vds.reference_data,
-        vds.variant_data.annotate_cols(
-            pop=meta_ht[vds.variant_data.col_key].population_inference.pop,
-            sex_karyotype=meta_ht[
-                vds.variant_data.col_key
-            ].sex_imputation.sex_karyotype,
-            fixed_homalt_model=meta_ht[
-                vds.variant_data.col_key
-            ].project_meta.fixed_homalt_model,
-            gatk_version=meta_ht[vds.variant_data.col_key].project_meta.gatk_version,
-            age=meta_ht[vds.variant_data.col_key].project_meta.age,
-            sample_age_bin=get_sample_age_bin(
-                meta_ht[vds.variant_data.col_key].project_meta.age
-            ),
+        vds.variant_data.select_cols(
+            pop=vds.variant_data.meta.population_inference.pop,
+            sex_karyotype=vds.variant_data.meta.sex_imputation.sex_karyotype,
+            fixed_homalt_model=vds.variant_data.meta.project_meta.fixed_homalt_model,
+            gatk_version=vds.variant_data.meta.project_meta.gatk_version,
+            age=vds.variant_data.meta.project_meta.age,
+            sample_age_bin=get_sample_age_bin(vds.variant_data.meta.project_meta.age),
             ukb_sample=hl.if_else(
-                meta_ht[vds.variant_data.col_key].project_meta.ukb_sample,
-                "ukb",
-                "non_ukb",
+                vds.variant_data.meta.project_meta.ukb_sample, "ukb", "non_ukb"
             ),
         ),
     )
 
+    # TODO: What above function? Also I think this can be done at a different time.
     # Downsamplings are done outside the above function as we need to annotate
```

Comment:
What above function?

Reply:
Annotation* but we can probably delete this comment.
```suggestion
    # Downsamplings are done outside the above annotation as we need to annotate
```